test/test_non_ssl.jl

```diff
@@ -1,140 +1,291 @@
-resp = ftp_put(ctxt, "test_upload.txt", file)
-@test resp.code ==226
-println("\nTest 10 passed.\n$(resp)")
-
-ftp_close_connection(ctxt)
-close(file)
-
-###############################################################################
-# Persistent connection tests, active mode
-###############################################################################
-
-options = RequestOptions(ssl=false, active_mode=true, username=user, passwd=pswd, hostname=host)
-println("\nTest persistent connection with active mode:\n")
-
-# test 11, establish connection
-ctxt, resp = ftp_connect(options)
-@test resp.code == 226
-println("\nTest 11 passed.\n$(resp)")
-
-# test 12, pass command to server
-resp = ftp_command(ctxt, "PWD")
-@test resp.code == 257
-println("\nTest 12 passed.\n$(resp)")
-
-# test 13, download file from server
-resp = ftp_get(ctxt, file_name)
-@test resp.code == 226
-println("\nTest 13 passed.\n$(resp)")
-# rm(file_name)
-
-# test 14, upload file to server
-file = open(upload_file)
-resp = ftp_put(ctxt, "test_upload.txt", file)
-@test resp.code ==226
-println("\nTest 14 passed.\n$(resp)")
-
-ftp_close_connection(ctxt)
-close(file)
-
-
-facts("Non-ssl tests") do
-
-save_file = "test_file_save_path.txt"
-save_path = pwd() * "/" * save_file
-
-context("download a file to a specific path") do
-
-    resp = ftp_get(file_name, options, save_path)
-    @fact resp.code --> 226
-    @fact isfile(save_file) --> true
-    file = open(save_file)
-    @compat @fact readstring(file) --> file_contents
-    close(file)
-    rm(save_file)
+@testset "FTPC.jl non ssl" begin
+    ftp_init()
+
+    non_ssl_test_upload = "test_upload.txt"
+
+    @testset "Non-persistent connection tests, passive mode" begin
+
+        options = RequestOptions(ssl=false, active_mode=false, username=user, passwd=pswd, hostname=host)
+        expected_header_first_part = AbstractString["220 Service ready for new user. (MockFtpServer 2.6; see http://mockftpserver.sourceforge.net)","331 User name okay, need password.","230 User logged in, proceed.","257 \"/\" is current directory."]
+        expected_header_port = r"229 Entering Extended Passive Mode \(\|\|\|\d*\|\)"
+
+        # The CI builds add this string to the end of the headers to non
+        # persistent connections.
+        possible_end_to_headers = "221 Service closing control connection."
+        @test options.url == "ftp://" * string(host) * "/"
+
+        @testset "ftp_get" begin
+            resp = ftp_get(file_name, options)
+            actual_body = readstring(resp.body)
+            @test resp.code == 226
+            @test typeof(resp.total_time) == Float64
+            @test resp.bytes_recd == file_size == length(actual_body)
+            @test actual_body == file_contents
+            expected_header_last_part = AbstractString["200 TYPE completed.","502 Command not implemented: SIZE.","150 File status okay; about to open data connection.","226 Closing data connection. Requested file action successful."]
```

Comment:
What would be the benefits of changing the type? It seems like a more complicated constructor.